contract.php

```diff
@@ -269,3 +269,52 @@ function contract_civicrm_post($op, $objectName, $id, &$objectRef){
     }
   }
 }
+
+/**
+ * Adjust output of membership custom data for contract file download
+ * @param $content
+ * @param $context
+ * @param $tplName
+ * @param $object
+ */
+function contract_civicrm_alterContent(  &$content, $context, $tplName, &$object ) {
```

Comment:
@systopia Also, I can't test if the file exists first if I use jquery.  So I would still need to hook into the form (through civicrm_buildForm), get the custom field and and assign it as a variable for use by jquery.  I didn't think I should spend too much time making it work so I went with the civicrm_alterContent.  Can you confirm that you'd like me to try the jquery method?